media_driver/linux/common/ddi/media_libva_util.cpp

```diff
@@ -354,6 +354,12 @@ VAStatus DdiMediaUtil_AllocateSurface(
                 gmmParams.BaseHeight = mediaSurface->pSurfDesc->uiSize / mediaSurface->pSurfDesc->uiPitches[0];
             }
         }
+        if(mediaSurface->pSurfDesc->uiVaMemType == VA_SURFACE_ATTRIB_MEM_TYPE_USER_PTR)
+        {
+            gmmParams.ExistingSysMemSize = mediaSurface->pSurfDesc->uiBuffserSize;
+            gmmParams.pExistingSysMem    = mediaSurface->pSurfDesc->ulBuffer;
+            gmmParams.Flags.Info.ExistingSysMem = true;
```

Comment:
Surprisingly this regress some use cases for me (where I use LibVA VP to copy surface from userptr allocated to driver allocated). Specifically YUY2 and UYVY regressed. And some cases which I anticipated will be fixed (some of i420, yv12) were not fixed.

YUY2 regressed in the following way. Input surface is 176x144 and I allocated it as follows:
1. 16-alignment for width and 64-bytes alignment for pitch. I.e. width aligned to 192, stride=384
2. height was 4-aligned and remained 144

With the above allocation YUY2 worked just fine before the change. After the change we fail on a call to CreateResInfoObject(). Basically gmmlib for some reason wants height to be 154 while it is eventually not. I am trying to clarify with gmmlib folks where this requirement comes from.

I420 case actually follows the same YUY2 failure pattern. Before the change it just was copied incorrectly.

Reply:
@dvrogozh have you synced with gmmlib folks about it? why the alignment required 154 instead 144?